src/plugins/pluginTypes.ts

```diff
@@ -35,6 +35,11 @@ export type IPlugin = {
   // simple regex replacements on the markdown output
   regexMarkdownModifications?: IRegexMarkdownModification[];
 
+  // operations on pages to define the markdown's frontmatter
+  frontmatterTransform?: {
+    build: (page: NotionPage) => string;

```

Comment:
how about `getFrontMatter()` to be parallel with `getStringFromBlock()` and `getReplacement()`.

Reply:
Recently I broke the plugin API when someone needed the `context: IDocuNotionContext,`. Although I don't know why it might be needed, let's just include it here, from the start. Please make it the 1st argument.